File: src/main/java/nodes/Visitor.java
package nodes;

import antlr.SCPBaseVisitor;
import antlr.SCPParser;
import nodes.BaseNode;
import nodes.BreakRule;
import nodes.expressions.*;
import nodes.statements.ForStatement;


import java.util.ArrayList;
import java.util.List;

public class Visitor extends SCPBaseVisitor<BaseNode> {
    public static List<String> code = new ArrayList<>();

    @Override
    public BaseNode visitFor_satement(SCPParser.For_satementContext ctx) {
        List<BaseNode> nodes = new ArrayList<>();
        for (int i = 0; i < ctx.iteration_body().code().size(); i++) {
            nodes.add(visit(ctx.iteration_body().code(i)));
        }

        ForStatement forStatement = new ForStatement(
                visitCondition(ctx.condition()),
                new Variable("int", new NameAndValue(ctx.iteration_variable().NODE(0).getText(), ctx.iteration_variable().NODE(1).getText())),
                visitIteration_change(ctx.iteration_change()), nodes);
        code.add(forStatement.toString());
        return forStatement;
    }

    @Override
    public Condition visitCondition(SCPParser.ConditionContext ctx) {
        return new Condition(ctx.role(0).NODE().getText(), ctx.role(1).NODE().getText(), visitOperators(ctx.operators()));
    }

    @Override
    public BaseNode visitBreak_rule(SCPParser.Break_ruleContext ctx) {
        BreakRule breakRule = new BreakRule();

        return breakRule;
    }

    @Override
    public Operator visitOperators(SCPParser.OperatorsContext ctx) {
        switch (ctx.getText()) {
            case "operator_less": {
                return new Operator("<");
            }
            case "operator_decrement": {
                return new Operator("--");
            }
        }
        return new Operator(ctx.getText());
    }


    @Override
    public IterationChange visitIteration_change(SCPParser.Iteration_changeContext ctx) {
        return new IterationChange(ctx.NODE(1).getText(), visitOperators(ctx.operators()));
    }

    @Override
    public BaseNode visitClass_statement(SCPParser.Class_statementContext ctx) {
        return super.visitClass_statement(ctx);
    }

}

File: src/main/kotlin/nodes/Visitor.kt
package nodes

import SCPBaseVisitor
import SCPParser.*
import nodes.expressions.*
import nodes.statements.ForStatement


class Visitor : SCPBaseVisitor<BaseNode>() {
    var code: MutableList<String> = ArrayList()

    override fun visitFor_satement(ctx: For_satementContext): BaseNode {
        val nodes: MutableList<BaseNode> = ArrayList()
        for (i in 0 until ctx.iteration_body().code().size) {
            nodes.add(visit(ctx.iteration_body().code(i)))
        }

        val forStatement = ForStatement(
            visitCondition(ctx.condition()),
            Variable("int", ctx.iteration_variable().NODE(1).text, NameAndValue(ctx.iteration_variable().NODE(0).text, "a")),
            visitIteration_change(ctx.iteration_change()), nodes
        )
        code.add(forStatement.toString())
        return forStatement
    }

    override fun visitCondition(ctx: ConditionContext): Condition {
//        return Condition(ctx.role(0).NODE().getText(), ctx.role(1).NODE().getText(), visitOperators(ctx.operators()))
        return Condition("a", "b", visitOperators(OperatorsContext(null, 1)))
    }

    override fun visitBreak_rule(ctx: Break_ruleContext?): BaseNode {
        return BreakRule()
    }

    override fun visitOperators(ctx: OperatorsContext): Operator {
        when (ctx.text) {
            "operator_less" -> {
                return Operator("<")
            }
            "operator_decrement" -> {
                return Operator("--")
            }
        }
        return Operator(ctx.text)
    }

    override fun visitIteration_change(ctx: Iteration_changeContext): IterationChange {
        return IterationChange(ctx.NODE(1).text, visitOperators(ctx.operators()))
    }

    override fun visitClass_statement(ctx: Class_statementContext?): BaseNode {
        return super.visitClass_statement(ctx)
    }

}